terraform/modules/splunk-server/resources.tf

```diff
@@ -40,7 +40,9 @@ resource "aws_instance" "splunk-server" {
       type        = "ssh"
       user        = "ubuntu"
       host        = aws_instance.splunk-server.public_ip
-      private_key = file(var.config.private_key_path)
+      agent       = var.config.use_ssh_agent == "1" ? true : false
```

Comment:
Why do we want to do this optional?
If key ssh key is not protected by password, will it work as usual? because if it will, we just add this code and remove the option in the configuration file.

Reply:
It would only work if the key had been added to ssh-agent

Reply:
so, if the password-less key hasn't been added to the agent it would fail if the flag is set?

Reply:
If use_ssh_agent=1 but the key hasn't been added to the agent then the terraform remote-exec connection will fail.